### PR TITLE
 EA-175, TEC-3301, TCMN-46: Update Upload mechanism of images

### PR DESCRIPTION
**Pull request by mitogh**
Update the mechanism to upload images into WordPress to allow larger file size of images and URLs that does not provide an extension on the image.

Tickets:

- [EA-175]
- [TEC-3301]
- [TCMN-46]

[EA-175]: https://moderntribe.atlassian.net/browse/EA-175
[TEC-3301]: https://moderntribe.atlassian.net/browse/TEC-3301
[TCMN-46]: https://moderntribe.atlassian.net/browse/TCMN-46

### CODE REVIEW

**Comment by lucatume** [comment]
I've updated the PR from the base branch and, implicitly, restarted the tests.

@bordoni I've added you as required reviewer too.

File: src/Tribe/Image/Uploader.php [hunk 1]
```diff
@@ -77,105 +113,138 @@ protected function upload_file( $file_url ) {
...
 
-		if ( false === $contents ) {
-			$message = sprintf( 'Could not upload image file "%s": failed getting the contents.', $file_url );
-			tribe( 'logger' )->log_error( $message, 'Image Uploader' );
+		$this->maybe_init_attachment_guids_cache();
+		$this->maybe_init_attachment_original_urls_cache();
 
-			return false;
-		}
+		self::$attachment_guids_cache[ get_post( $attachment_id )->guid ] = $attachment_id;
```
Comment (bordoni) [inline src/Tribe/Image/Uploader.php]: We should save this into a variable and check before using it.

File: src/Tribe/Image/Uploader.php [hunk 2]
```diff
@@ -77,105 +113,138 @@ protected function upload_file( $file_url ) {
...
+	 * @return string Returned a file name with an extension if is not already part of the file name.
+	 */
+	protected function create_file_name( $filename ) {
+		/**
 		 * We use the path basename only here to provided WordPress with a good filename
 		 * that will allow it to correctly detect and validate the extension.
 		 */
-		$path   = parse_url( $file_url, PHP_URL_PATH );
-		$upload = wp_upload_bits( basename( $path ), null, $contents );
+		$path = parse_url( $filename, PHP_URL_PATH );
```
Comment (bordoni) [inline src/Tribe/Image/Uploader.php]: Use `wp_parse_url`

File: readme.txt [hunk 3]
```diff
@@ -5,6 +5,7 @@
 = [TBD] TBD =
 
 * Added context to the country and the state of Georgia to allow separate translation [BTRIA-63]
+* Allow uploads of images with a large size and images with no extension provided from the URL, as the extension from the URL was used to define the type of the file to be uploaded and when the extension was not present on the URL the file was considered invalid. [TCMN-46]
```
Comment (bordoni) [inline readme.txt]:
```suggestion
* Tweak - Allow uploads of images with a large size and images with no extension provided from the URL, as the extension from the URL was used to define the type of the file to be uploaded and when the extension was not present on the URL the file was considered invalid. [TCMN-46]
```

File: src/Tribe/Image/Uploader.php (hunk 1 above)
Comment (mitogh) [reply src/Tribe/Image/Uploader.php]: We have a check for this a little bit above.

Here: https://github.com/moderntribe/tribe-common/blob/eeac34c30b926568356eeff5c18c5fe8f3f43c7d/src/Tribe/Image/Uploader.php#L162

**Pull request merged**